### ISSUE TITLE
completion menu opened twice

### ISSUE THREAD

**Issue by lacygoill** [vim/vim#12422]
**Steps to reproduce**

Run this shell command:

    vim -Nu NONE -S <(tee <<'EOF'
        vim9script

        &linebreak = true
        &showbreak = '+ '
        &signcolumn = 'yes'
        &wrap = true

        var yyy = 'yyyyyyyyyyyyyyyyyyyy'
        (['x'->repeat(2 * &columns - yyy->strlen() - 15) .. ' ' .. yyy, ''] + ['x']->repeat(10))->setline(1)

        g:matches =<< trim END
            zzzzzzzzzzzzzzz.zzzzzzz
            zzzzz.zzzz
            zzzzzz.zzzzzzz
            zzzzzzzzz.zzzzzz.zzzzz
            zzzzzzz.zz.zzzzzzzzzzzzzzz.zzzzzzz
            zzzzzzzz.zzzzz.zzzzzzz
            zzzzzzzz.zzzzzzzzzzz.zzzzz.zzzzzzz
            zzzzzzzz.zzzzzzzzzzz.zzzzzzzzzzzzz.zzzzzzz
            zzzzzzzz.zzzzzzzzzzz.zzzzzzzzzzzzz.zzzzzzz
            zzzzzzzz.zzzzzzzzzzz.zzzzzzzz.zzzzzzz
        END
        inoremap <Tab> <C-R>=complete(col('.'), g:matches) ? '' : ''<CR>
        startinsert!
        feedkeys("\<Tab>\<C-N>")
    EOF
    )

The completion menu is drawn in two different locations.

**Expected behavior**

The completion menu is drawn in only one single location.

**Version of Vim**

9.0 Included patches: 1-1572

**Environment**

Operating system: Ubuntu 20.04.6 LTS
Terminal: XTerm(378)
Value of $TERM: xterm-256color
Shell: GNU bash, version 5.0.17

**Video**

https://github.com/vim/vim/assets/8505073/c0d88589-9776-4b48-957c-652d9d235816

**Comment by lacygoill** [comment]
No, I didn't migrate.  But I had too many repositories.  It was taking too much time to version control them.  So I moved most of them into my config.  I made the latter private because there are certain parts which I don't want to be public.  And having to maintain 2 separate repos (a public one + a private one) seemed too cumbersome.

**Comment by brammool** [comment]
It can be fixed by removing the last "if" in ins_compl_upd_pum(). 

That is not the right solution though, the condition should be adjusted in some way.  Otherwise I suspect the menu will flicker.  Probably the position of the menu needs to be taken into account, but it's only computed later.

The reproduction looks complicated.  Can you reduce it, so that it can be used for a regression test?

**Comment by lacygoill** [comment]
> The reproduction looks complicated. Can you reduce it, so that it can be used for a regression test?

OK, I tried to reduce further:
```vim
vim9script
&linebreak = true
&showbreak = '+ '
&signcolumn = 'yes'
&wrap = true
var y = 'y'->repeat(20)
(['x'->repeat(2 * &columns - y->strlen() - &pumwidth) .. ' ' .. y, ''] + ['x']->repeat(10))->setline(1)
inoremap <Tab> <ScriptCmd>complete(col('.'), ['zzzzzzzzzzzzzzz.zzzzzzz', 'zzzzz.zzzz'])<CR>
startinsert!
feedkeys("\<Tab>\<C-N>")
```

**Comment by lacygoill** [comment]
Yet a little simpler than the previous one:
```vim
vim9script
&linebreak = true
&showbreak = '+ '
&signcolumn = 'yes'
&wrap = true
var y = 'y'->repeat(20)
(['x'->repeat(2 * &columns - y->strlen() - &pumwidth) .. $' {y}', ''])->setline(1)
inoremap <Tab> <ScriptCmd>complete(col('.'), ['zzzzzzzzzzzzzzz.zzzzzzz', 'zzzzz.zzzz'])<CR>
feedkeys("A\<Tab>\<C-N>")
```

**Comment by lacygoill** [comment]
I don't think I can reduce more than this one:
```vim
vim9script
&linebreak = true
&wrap = true
var y = 'y'->repeat(20)
(['x'->repeat(2 * &columns - y->strlen() - &pumwidth) .. $' {y}', ''])->setline(1)
inoremap <F3> <ScriptCmd>complete(col('.'), [y, 'zzzzzzz.zzzzzzz'])<CR>
feedkeys("A\<F3>\<C-N>")
```
